### PR TITLE
Add pyarrow to moose-tools

### LINKED ISSUE #27018: Add Pyarrow to moose-tools package
### Motivation

The Pandas Python package within the current `moose-tools` is now presenting a deprecation warning when scripts are using it: 

```
DeprecationWarning:
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
```

### Design

Add pyarrow to the `moose-tools` conda package.

### Impact

Avoidance of future breakages when using pandas.

### PR DESCRIPTION
**Pull request by cticenhour**
Closes #27018

### CODE REVIEW

File: framework/doc/packages_config.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ minimum_python: 3.7
 mpich: 4.0.2
 petsc_alt: 3.11.4
 petsc: 3.20.3
-moose_dev: 2024.03.05
-moose_tools: 2023.12.20
+moose_dev: 2024.03.07
+moose_tools: 2024.03.07
```
Comment (milljm) [inline framework/doc/packages_config.yml:10]: way to be!

**Review by milljm — APPROVED** [review]
👍

File: conda/tools/meta.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ requirements:
     - setuptools
   host:
     # TODO: Try to remove when this version settles down among other conda-forge packages
-    - openssl <3.2.0a # [osx]
+    - openssl <3.1.5 # [osx]
```
Comment (cticenhour) [inline conda/tools/meta.yaml:30]: @milljm Making this change fixed up the dependency issues seen on CIVET with `moose-dev`.

**Review by milljm — APPROVED** [review]
👍

**Pull request merged**